ui/src/components/admin/sermonNotes/SermonNotesEditorModal.js

```diff
@@ -107,6 +107,32 @@ const SermonNotesEditorModal = (props) => {
     }
   }, [toast]);
 
+  const checkDuplicateTitleAndSermonLink = async () => {
+    try {
+      const data = await fetchSermonNotes();
```

Comment:
you shouldn't be fetching the data again here, since you already fetch it in the initial mount. Ideally we should minimize the API calls to reduce network.